src/main/java/seedu/address/model/person/Course.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's course in the Intern Watcher.
+ * Guarantees: immutable; is valid as declared in {@link #isValidCourse(String)}
+ */
+public class Course {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Course Names should only contain alphanumeric characters and spaces, and it should not be blank";
+
+    /*
+     * The first character of the course must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
```

Comment:
Maybe Just letters for the course would be better as I dont think courses have numbers(Might be wrong)

Reply:
Yup it makes sense. I'll change it accordingly.